backend/census_historical_migration/workbooklib/secondary_auditors.py

```diff
@@ -94,6 +93,25 @@ def _get_secondary_auditors(dbkey, year):
     return sort_by_field(results, "ID")
 
 
+def xform_cpafirmname(secondary_auditors):
+    change_records = []
+    is_empty_cpafirmname_found = False
+    for secondary_auditor in secondary_auditors:
+        if secondary_auditor.CPAFIRMNAME == "":
+            is_empty_cpafirmname_found = True
+            track_transformations(
```

Comment:
We want to call `track_transformations` regardless of `secondary_auditor.CPAFIRMNAME == ""` condition. However, we want save the record (i.e., `InspectionRecord.append_secondary_auditor_changes(change_records)`) only if `is_empty_cpafirmname_found` is true.

Reply:
could we clarify why this is the case as a comment in the code for future generations?

Reply:
This is the convention we agreed upon before the first round of migration. For context, when a report section undergoes a transformation, the transformation function reviews each record in that section and modifies those that require transformation. At the end of the section, we need to determine which records were modified and which were not. Because we do not migrate any database primary keys along with the records, it becomes very challenging—and in some cases, nearly impossible—to accurately inform users which records have changed and which have not on the dissemination side. This is  because we lack a way to directly link each record in the `inspectionrecord` table to its corresponding record in the dissemination tables.
By saving all the records from the `inspectionrecord` for the given section, we can use the order of appearance in both the `inspectionrecord` and dissemination table records to establish the aforementioned link between records in `inspectionrecord` table and records in dissemination tables.

Also, We only save changes into the `inspectionrecord` table for a given section of a report if at least one record from that section has been modified.